### DISCUSSION TITLE
Wrong image reported in container status

### DISCUSSION THREAD

**Discussion by lnthai2002** [arkmq-org/arkmq-org-broker-operator#1042] [Q&A] [answered]
Hello, i am using io.fabric8:kubenretes-client java client to talk to my kubernetes cluster to get status and image from as artemis pod deployed by artemis-operator 1.2.1. However, in the containerStatuses section, the image reported is different from image reported using kubectl.  Attached is the raw json returned when my java client asked k8s for all info related to artemis pod. Please pay attention to lines 1227-1247 and 1862-1869. I am not sure if this is an issue with k8s of the artemis-operator, hope someone can shed some light on this.

Regards
Thai Le

[pod_info.json](https://github.com/user-attachments/files/17546162/pod_info.json)

**Comment by brusdev** [comment]
@lnthai2002 the image in [pod_info.json](https://github.com/user-attachments/files/17546162/pod_info.json) is `quay.io/artemiscloud/activemq-artemis-broker-kubernetes@sha256:b293c1269166cc6181f68b8d803fdfcf20fdbd5df2787ed0453724b22e298fb2` and that is expected.

What different container image do you see?

**Reply by lnthai2002** [reply]
In the json file i sent, line 1863 shows that the image of the "activemq-artemis-broker-container" is "sha256:039815d92b2becd39601d3871d08cc8692820756801d5970556128bb51c8d2dc".
![image](https://github.com/user-attachments/assets/86dff52b-7675-43fa-8d71-4f42d726686b)
This "image" field always report correct image tag for all of our pods except artemis.

**Reply by brusdev** [reply, marked as answer]
The container image/imageId may not match the image used in the PodSpec, as it may have been resolved by the runtime, see https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.28/#containerstatus-v1-core

Indded there is also a related kubernetes issue: https://github.com/kubernetes/kubernetes/issues/115199

Furthermore `quay.io/artemiscloud/activemq-artemis-broker-kubernetes@sha256:b293c1269166cc6181f68b8d803fdfcf20fdbd5df2787ed0453724b22e298fb2` points to a container image manifest with multiple architectures, see https://developers.redhat.com/articles/2023/11/03/how-build-multi-architecture-container-images

**Reply by lnthai2002** [reply]
Thank you for the links, this multi-arch manifest is new to me, I gonna do some research on this.
